lib/features.js

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+module.exports = config => {
+  const lastAccessTimeUpdates = config.lastAccessTimeUpdates
+  const signinConfirmation = config.signinConfirmation
+
+  return {
+    isSampledUser: isSampledUser,
+
+    isLastAccessTimeEnabled (uid, email) {
+      return lastAccessTimeUpdates.enabled && (
+        isSampledUser(uid, lastAccessTimeUpdates.sampleRate) ||
+        new RegExp(lastAccessTimeUpdates.enabledEmailAddresses).test(email)
+      )
+    },
+
+    isSigninConfirmationEnabled (uid, email, request) {
```

Comment:
Here's your refactored function @vbudhram. It is exactly the same except:

* The name has changed slightly.
* It's been *es6ialized*.
* I wrote some unit tests for it because, unless I've missed them, there weren't any.